x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/action/ReindexDataStreamIndexAction.java

```diff
@@ -0,0 +1,141 @@
+
+    private ReindexDataStreamIndexAction() {
+        super(NAME);
+    }
+
+    public static class Request extends ActionRequest implements IndicesRequest {
+
+        private final String sourceIndex;
+        private final boolean deleteDestIfExists;
+
+        public Request(String sourceIndex) {
+            this(sourceIndex, true);
+        }
+
+        public Request(String sourceIndex, boolean deleteDestIfExists) {
+            this.sourceIndex = sourceIndex;
+            this.deleteDestIfExists = deleteDestIfExists;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.sourceIndex = in.readString();
+            this.deleteDestIfExists = in.readBoolean();
+        }
+
+        public static Request readFrom(StreamInput in) throws IOException {
+            return new Request(in);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(sourceIndex);
+            out.writeBoolean(deleteDestIfExists);
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        public String getSourceIndex() {
+            return sourceIndex;
+        }
+
+        public boolean getDeleteDestIfExists() {
+            return deleteDestIfExists;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request request = (Request) o;
+            return Objects.equals(sourceIndex, request.sourceIndex) && deleteDestIfExists == request.deleteDestIfExists;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(sourceIndex, deleteDestIfExists);
+        }
+
+        @Override
+        public String[] indices() {
+            return new String[] { sourceIndex };
+        }
+
+        @Override
+        public IndicesOptions indicesOptions() {
+            return IndicesOptions.strictSingleIndexNoExpandForbidClosed();
+        }
+    }
+
+    public static class Response extends ActionResponse {
+        private final String destIndex;
+        private final long numCreated;
+
+        public Response(String destIndex, long numCreated) {
+            this.destIndex = destIndex;
+            this.numCreated = numCreated;
+        }
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+            this.destIndex = in.readString();
+            this.numCreated = in.readLong();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(destIndex);
+            out.writeLong(numCreated);
+        }
+
+        public String getDestIndex() {
+            return destIndex;
+        }
+
+        public long getNumCreated() {
```

Comment:
What is numCreated meant to be? What should a client do with this?

Reply:
Yeah ... it's only used in tests to confirm that if the dest index is not deleted, only previously un-added docs are copied over. Think I should remove it?